DEVELOPMENT_BEST_PRACTICE.md

```diff
@@ -0,0 +1,308 @@
+
+## Defining _"clean code that is optimized for change"_
+
+What constitutes clean code? We want simple and maintainable, but how do we define that? The following best practices offer niche advice. Collectively, it represent our definition of clean code.
+
+### State
+
+> Shared mutable state is believed by many to be the “root of all evil”, or at least the cause of most of the accidental complexity in our code. And “Complexity is the root cause of the vast majority of problems with software today.” - [Mauro Bieg](https://mb21.github.io/blog/2021/01/23/pure-functional-programming-and-shared-mutable-state.html#the-root-of-all-evil)
+
+Package level state is rarely required. If it is, it should be composed of re-usable constructs defined and exposed from the package itself.
+
+In avoiding package level state we avoid [hidden dependencies and unintended side effects](https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_package_level_state) stemming from global state mutation. Model loosely-coupled components by [declaring your dependencies](#dependencies).
+
+### Names
+
+##### Packages
+
+Package names help form the first impression of what a package provides and is used to reference the public API of the package. A good package name brings about clarity and purpose for consumers. Package names should be concise and represent the behavior the package provides not the types it contains. 
+
+Avoid stutter in struct and interface names by treating the package name as a namespace and considering how the types are consumed. For example, `drink.DrinkDecorator` should be `drink.Decorator`. You may find you have a type named after the package, soemtimes it's appropriate.
+
+If you find yourself using an appropriate package name that's commonly used as a variable, consider pluralising the package name.
+
+Avoid [relentlessly] `types`, `interfaces`, `common`, `util` or `base` packages. These names offer no indication on what the package provides beyond 'a bundle of stuff'. With time, these types of package turn into dumping grounds containing unrelated logic that can be the root cause for cyclic dependencies.  If a utility style package is required, make it specific. For example a `/utils/cmdutil` package may provide command utility functions.
+
+##### Function & method
+
+Functions and methods (functions with a receiver parameter) should adequetly describe their behavior. Generally, they should follow a verb-noun form. The types of parameters should help inform the function name.
+
+##### Variable
+
+Variable names should be concise and descriptive. Prefer single word names. The further away from the site of declaration a variable is used, the more descriptive it needs to be. For example, a variable named `n` used 30 lines after declaration makes it unnecessarily difficult to reason what it represents at the site of use.
```

Comment:
Is this doc serving for general Go practice or EKS Anywhere best practice?

If latter, can we expand a bit for recommended eks-a naming convention? e.g. long vs short:`kubeadmControlPlane` vs `kcp`; capitalize acronym or not: `capi` vs `CAPI`, `ip` vs `IP`, etc.

Reply:
Its both but its scope only extends to EKS-A :). 

Re naming, we can expand the variable naming section to elaborate on recommendations. We should probably end with 'if in doubt, prefer clarity over brevity'.

Reply:
Addressed.